server/service/hosts.go

```diff
@@ -545,22 +556,30 @@ func refetchHostEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 }
 
 func (svc *Service) RefetchHost(ctx context.Context, id uint) error {
-	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
-		return err
+	// can be already authorized if coming from device auth token endpoint
+	var alreadyAuthd bool
+	if authctx, ok := authzctx.FromContext(ctx); ok {
+		alreadyAuthd = authctx.Checked()
```

Comment:
Maybe we can abstract this logic somehow? (Seems that the logic will repeat on every API used by both users and hosts.)

Reply:
yeah I guess I could create a `authzctx.IsAlreadyAuthenticated(ctx) bool` function in that package and use that to make the code clearer. Did you have something more or a completely different approach in mind?

Reply:
I thought of https://github.com/fleetdm/fleet/pull/4531/files/b9200d182bca1342c39057ac5ae80efd31ec9317#r822900766 but now I see your response. Let's continue the discussion there.